include/mbedtls/bignum.h

```diff
@@ -880,7 +880,40 @@ int mbedtls_mpi_mod_int(mbedtls_mpi_uint *r, const mbedtls_mpi *A,
                         mbedtls_mpi_sint b);
 
 /**
- * \brief          Perform a sliding-window exponentiation: X = A^E mod N
+ * \brief          Perform a modular exponentiation: X = A^E mod N
+ *
+ * \warning        This function is not constant time with respect to \p E (the exponent).
+ *
+ * \param X        The destination MPI. This must point to an initialized MPI.
+ *                 This must not alias E or N.
+ * \param A        The base of the exponentiation.
+ *                 This must point to an initialized MPI.
+ * \param E        The exponent MPI. This must point to an initialized MPI.
+ * \param N        The base for the modular reduction. This must point to an
+ *                 initialized MPI.
+ * \param prec_RR  A helper MPI depending solely on \p N which can be used to
+ *                 speed-up multiple modular exponentiations for the same value
+ *                 of \p N. This may be \c NULL. If it is not \c NULL, it must
+ *                 point to an initialized MPI. If it hasn't been used after
+ *                 the call to mbedtls_mpi_init(), this function will compute
+ *                 the helper value and store it in \p prec_RR for reuse on
+ *                 subsequent calls to this function. Otherwise, the function
+ *                 will assume that \p prec_RR holds the helper value set by a
+ *                 previous call to mbedtls_mpi_exp_mod(), and reuse it.
+ *
+ * \return         \c 0 if successful.
+ * \return         #MBEDTLS_ERR_MPI_ALLOC_FAILED if a memory allocation failed.
+ * \return         #MBEDTLS_ERR_MPI_BAD_INPUT_DATA if \c N is negative or
+ *                 even, or if \c E is negative.
+ * \return         Another negative error code on different kinds of failures.
+ *
+ */
+int mbedtls_mpi_exp_mod_unsafe(mbedtls_mpi *X, const mbedtls_mpi *A,
```

Comment:
I'm sorry but I realized I made a mistake in a [previous approval](https://github.com/Mbed-TLS/mbedtls/pull/9281#pullrequestreview-2236094854):

> Since it's not in the public API, we can look for a better way after 3.6.1.

But as of 878af1280729c60c4fb965005fae929963b0c20d we _are_ adding `mbedtls_mpi_exp_mod_unsafe` to the public API of the 3.6 LTS branch. I'm [ok with the prototype](https://github.com/Mbed-TLS/mbedtls/pull/9281#discussion_r1666676645) but I [don't like the name](https://github.com/Mbed-TLS/mbedtls/pull/9281#issuecomment-2291335763). Also I don't like to rush when extending the API of an LTS branch.

Can we make this an internal function instead? It's only meant to be used in `rsa.c` after all.

Reply:
If you don't like the name we need to have a larger discussion because we already have a few functions called `_unsafe` (as Janos pointed out already), so if we make a change it should apply to all functions.

But I fully agree about making the new function internal, especially considering it's an LTS and we don't have time to have the naming discussion.

Reply:
> we already have a few functions called _unsafe

Not in the public API. I don't care nearly as much about internal functions.

```
$ git show -s --oneline
28cdd11908 (HEAD, upstream-restricted/test-mbedtls-3.6.1rc, upstream-public/mbedtls-3.6) Merge pull request #9479 from gilles-peskine-arm/psa-keystore-static-release-update-3.6
$ git grep _unsafe include
```

We do have one internal function called `xxx_unsafe` (`mbedtls_mpi_core_get_mont_r2_unsafe`) and while I'm ok (but not enthusiastic) about the name, I'm bothered that its documentation doesn't explain in what way it's unsafe.

Reply:
Indeed, the other function is not public, so that's an additional reason to keep the new one private.

(I also agree that naming is less important for internal functions, but we can still have a discussion about it some time. I can see arguments for calling them `_leaky` rather than `_unsafe` as it's more informative. Also, for the small `static` functions that are currently called `optionally_safe`, I think `slow_or_leaky` would make anyone think before using them. But let's have this discussion another time.)